test/utils/setup.ts

```diff
@@ -7,6 +7,16 @@ import { safeContractUnderTest } from "./config";
 import { getRandomIntAsString } from "./numbers";
 import { Safe, SafeL2, SafeMigration } from "../../typechain-types";
 
+type safeWithFullConfig = {
+    readonly owners: string[];
+    threshold?: number;
```

Comment:
should other be readonly as well?

Reply:
I'd split the type for better semantics: e.g.,
```
type SafeSetupConfig = { ...all the .setup function parameters }
type SafeCreationConfig = SafeSetupConfig & { logGasUsage?: boolean }
```
or
```
type SafeSetupConfig = { ...all the .setup function parameters }
type WithLogging<T> = T & { logGasUsage?: boolean }
type SafeCreationConfig = WithLogging<SafeSetupConfig>
```
or
```
type SafeSetupConfig = { ...all the .setup function parameters }
type ParamsWithGasLogs = { logGasUsage?: boolean }
type SafeCreationConfig = SafeSetupConfig & ParamsWithGasLogs
```

Reply:
I think they can all be `readonly`.

Reply:
Personally, I don't know if splitting the types here necessarily makes sense - since we don't use any of the other subtypes anywhere. To me its just additional cognitive load without any real benefit.

Reply:
> should other be readonly as well?

So my initial thoughts on this were:

> The owners parameter is kept as readonly because it is not supposed to change within the function, the rest of all the parameters in the type could be changed based on if any value is passed or not (if undefined, it will be assigned a default value).

Reply:
> Personally, I don't know if splitting the types here necessarily makes sense - since we don't use any of the other subtypes anywhere. To me its just additional cognitive load without any real benefit.

I can see the possibility of a developer wanting to define a safe configuration object in the code and for that, they may want to import a type, but currently, our type contains irrelevant properties, like `logGasUsage` (not required but still)

Reply:
Segregated 👍🏾